systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/AbstractKafkaClient.java

```diff
@@ -4,21 +4,20 @@
  */
 package io.strimzi.systemtest.kafkaclients;
 
-import io.fabric8.kubernetes.api.model.LoadBalancerIngress;
-import io.fabric8.kubernetes.api.model.Service;
-import io.fabric8.openshift.api.model.Route;
-import io.fabric8.openshift.client.OpenShiftClient;
+import io.strimzi.api.kafka.model.status.ListenerStatus;
+import io.strimzi.systemtest.kafkaclients.clientproperties.ConsumerProperties;
+import io.strimzi.systemtest.kafkaclients.clientproperties.ProducerProperties;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
 import io.strimzi.systemtest.utils.ClientUtils;
 import org.apache.kafka.common.security.auth.SecurityProtocol;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import java.security.InvalidParameterException;
+import java.util.List;
 
-import static io.strimzi.api.kafka.model.KafkaResources.externalBootstrapServiceName;
-import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
-
-public abstract class AbstractKafkaClient {
+@SuppressWarnings("unchecked")
```

Comment:
I'm guessing this is because the fluent pattern on the builder with each method having to cast `(SELF) this` results in lots of warnings. It is best to use `@SuppressWarnings` on the smallest scope possible, so that genuine warnings are not suppressed accidentally. But you understandably don't want to have to annotate each method, so the solution is to write a method `self()` does the cast and is annotated, then all the rest get to `return self();` and not needing annotating. Basically the rule of thumb is always to refactor you untypesafe code into a method and annotate just that method.

Reply:
Yeah, make sense thanks.